Adding codeowners to quantization, sparsity, ns, etc.

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #79505

The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

Commit:
Adding codeowners to quantization, sparsity, ns, etc.

The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

[ghstack-poisoned]

Commit:
Update on "Adding codeowners to quantization, sparsity, ns, etc."


The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

[ghstack-poisoned]

jerryzh168 commented on CODEOWNERS:
```diff
@@ -16,6 +16,19 @@
+nn/quantizable/ @jerryzh168 @z-a-f
+nn/qat/ @jerryzh168 @vkuzo
+nn/intrinsic @jerryzh168 @vkuzo
+ao/ @vkuzo
```
can you add ao/quantization and add my name as well

jerryzh168 replied:
and why this does not start with `/`?

albanD replied:
>and why this does not start with /?

This is the same as `**/ao/*` basically. So anything that has `/ao/` in its path will get added.

z-a-f commented:
@mergebot rebase

z-a-f commented:
@pytorchbot rebase

Commit:
Update on "Adding codeowners to quantization, sparsity, ns, etc."


The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

[ghstack-poisoned]

Commit:
Update on "Adding codeowners to quantization, sparsity, ns, etc."


The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

[ghstack-poisoned]

jerryzh168 reviewed (commented):
I think we want to add ownership for https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized as well, can have separate owners for:
```
https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized
https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized/cpu
https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized/cuda
https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized/cudnn
```

z-a-f commented:
> I think we want to add ownership for https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized as well, can have separate owners for:
> 
> ```
> https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized
> https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized/cpu
> https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized/cuda
> https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/quantized/cudnn
> ```

Thank you very much! Will update. Please, let me know if I missed any other places?

Commit:
Update on "Adding codeowners to quantization, sparsity, ns, etc."


The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

[ghstack-poisoned]

jerryzh168 commented:
I guess there is a parallel place for sparsity as well: https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/ao_sparse

jerryzh168 commented:
also is it possible to infer the codeowners by some heuristics?

Commit:
Update on "Adding codeowners to quantization, sparsity, ns, etc."


The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

[ghstack-poisoned]

Commit:
Update on "Adding codeowners to quantization, sparsity, ns, etc."


The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

[ghstack-poisoned]

Commit:
Update on "Adding codeowners to quantization, sparsity, ns, etc."


The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

[ghstack-poisoned]

Commit:
Update on "Adding codeowners to quantization, sparsity, ns, etc."


The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

[ghstack-poisoned]

z-a-f commented:
@pytorchbot rebase

Commit:
Update on "Adding codeowners to quantization, sparsity, ns, etc."


The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

[ghstack-poisoned]

z-a-f commented on CODEOWNERS:
```diff
@@ -21,6 +21,25 @@
+/aten/src/ATen/native/ao_sparse @z-a-f @salilsdesai @kimishpatel @digantdesai
+/aten/src/ATen/native/quantized @vkuzo @jerryzh168
+/aten/src/ATen/native/quantized/cpu @vkuzo @jerryzh168
+/aten/src/ATen/native/quantized/cuda @vkuzo @jerryzh168
+/aten/src/ATen/native/quantized/cudnn @vkuzo @jerryzh168
```
Add David here

z-a-f commented on CODEOWNERS:
```diff
@@ -21,6 +21,25 @@
+# Architecture Optimization (quantization, sparsity, etc.)
+/aten/src/ATen/native/ao_sparse @z-a-f @salilsdesai @kimishpatel @digantdesai
+/aten/src/ATen/native/quantized @vkuzo @jerryzh168
+/aten/src/ATen/native/quantized/cpu @vkuzo @jerryzh168
```
Add FBGEMM and QNNPACK here

Commit:
Update on "Adding codeowners to quantization, sparsity, ns, etc."


The notifications for the AO-maintained codebase.
This should not be blocking, just PR/test notifications.

[ghstack-poisoned]

z-a-f commented:
@pytorchbot merge -g
